src/api-binder.ts

```diff
@@ -342,13 +342,13 @@ export class APIBinder {
 		if (deviceId == null) {
 			throw new Error('Attempt to retrieve device tags before provision');
 		}
-		const tags = (await this.balenaApi.get({
+		const tags = await this.balenaApi.get({
 			resource: 'device_tag',
 			options: {
 				$select: ['id', 'tag_key', 'value'],
 				$filter: { device: deviceId },
 			},
-		})) as Array<Dictionary<unknown>>;
```

Comment:
The primary changes in the update are improved typings, which is what allows us to drop some of these casts